packages/astro/src/@types/astro-core.ts

```diff
@@ -139,8 +139,12 @@ export interface CollectionRSS {
 
 /** Generic interface for a component (Astro, Svelte, React, etc.) */
 export interface ComponentInstance {
+  $$metadata: {
+    modules: { module: ComponentInstance; specifier: string }[];
```

Comment:
`module` isn't a ComponentInstance. It's the namespace of everything you import (so like `import * as mod from 'foo';`). Probably this should be `Record<string, any>`